### DISCUSSION TITLE
Will the map rotation feature be included in version 0.22? When is the new version expected to be released?

### DISCUSSION THREAD

**Discussion by ekinkaradagg** [mapsforge/mapsforge#1532] [Q&A] [answered]
I have a project where it's important to include the map rotation feature, so I wanted to get some information about it. Thank you very much for your efforts!

**Comment by devemux86** [comment, marked as answer]
[Map rotation](https://github.com/mapsforge/mapsforge/issues/7#issuecomment-2147721066) has been implemented, it will be available in 0.22 very soon.

You can already use it with the `master-SNAPSHOT` version via JitPack (see instructions [here](https://github.com/mapsforge/mapsforge/blob/master/docs/Integration.md)).
[RotationMapViewer](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/src/main/java/org/mapsforge/samples/android/RotationMapViewer.java) demonstrates the new rotation API.

<img src="https://github.com/cgeo/cgeo/assets/3484020/f512219e-362a-49d1-9ead-26edd877d845" width="50%"/><img src="https://github.com/cgeo/cgeo/assets/3484020/d36dfcb1-c8da-469c-883a-419333013f84" width="50%"/>

**Reply by ekinkaradagg** [reply]
Thank you so much for your efforts! When the rotation angle increases, street and location names can appear at an inverted angle and become unreadable. Is this my mistake or a general issue?

**Reply by devemux86** [reply]
This is a native map rotation:
- Only the underlying map is rotated, not the labels / symbols
- POI labels / symbols are always horizontally facing upwards
- Path texts are automatically rotated to be readable

This works similarly to all vector map libraries that support native map rotation.

---

You can also try it in Cruiser 5 navigation and follow modes (long press location button):
- https://github.com/devemux86/cruiser/discussions/610
